js/modules/k6/ws/ws.go

```diff
@@ -155,17 +155,11 @@ func (mi *WS) Connect(url string, args ...goja.Value) (*WSHTTPResponse, error) {
 					header.Set(key, headersObj.Get(key).String())
 				}
 			case "tags":
-				tagsV := params.Get(k)
-				if goja.IsUndefined(tagsV) || goja.IsNull(tagsV) {
-					continue
-				}
-				tagObj := tagsV.ToObject(rt)
-				if tagObj == nil {
-					continue
-				}
-				for _, key := range tagObj.Keys() {
-					tags = tags.With(key, tagObj.Get(key).String())
+				newTags, err := common.ApplyCustomUserTags(rt, tags, params.Get(k))
+				if err != nil {
+					return nil, err
```

Comment:
Not sure what the end user-faced error looks like, but if it's generic, I'd also suggest explicitly mentioning that this is coming from the WebSockets module. And basically it's relevant for any place like HTTP, groups, or whatever.

My point if we explicitly say "where" the issue is to the users, they will be happy :relaxed: